apps/prairielearn/src/pages/instructorStudentDetail/instructorStudentDetail.html.tsx

```diff
@@ -72,210 +67,41 @@ export function InstructorStudentDetail({
-            <button type="button" class="btn btn-sm btn-light" onClick={handleViewAsStudent}>
-              <i class="fas fa-user-graduate me-1" aria-hidden="true" />
-              View as student
-            </button>
-          )}
-        </div>
-        <div class="card-body">
-          <div class="d-flex align-items-center justify-content-between">
-            <h2 class="d-flex align-items-center">
-              {user?.name ?? enrollment.pending_uid}
-              {enrollment.status !== 'joined' && (
-                <EnrollmentStatusIcon type="badge" class="ms-2 fs-6" status={enrollment.status} />
-              )}
-            </h2>
-            {hasCourseInstancePermissionEdit && enrollmentManagementEnabled && (
-              <div class="d-flex gap-2 align-items-center">
-                {enrollment.status === 'joined' && (
-                  <form method="POST">
-                    <input type="hidden" name="__csrf_token" value={csrfToken} />
-                    <input type="hidden" name="__action" value="block_student" />
-                    <button type="submit" class="btn btn-sm btn-outline-danger">
-                      <i class="fas fa-user-slash me-1" aria-hidden="true" /> Block student
-                    </button>
-                  </form>
-                )}
-                {enrollment.status === 'blocked' && (
-                  <form method="POST">
-                    <input type="hidden" name="__csrf_token" value={csrfToken} />
-                    <input type="hidden" name="__action" value="unblock_student" />
-                    <button type="submit" class="btn btn-sm btn-outline-success">
-                      <i class="fas fa-user-check me-1" aria-hidden="true" /> Remove block
-                    </button>
-                  </form>
-                )}
-                {enrollment.status === 'invited' && (
-                  <form method="POST">
-                    <input type="hidden" name="__csrf_token" value={csrfToken} />
-                    <input type="hidden" name="__action" value="cancel_invitation" />
-                    <button type="submit" class="btn btn-sm btn-outline-secondary">
-                      <i class="fas fa-times me-1" aria-hidden="true" /> Cancel invitation
-                    </button>
-                  </form>
-                )}
-                {enrollment.status === 'rejected' && (
-                  <form method="POST">
-                    <input type="hidden" name="__csrf_token" value={csrfToken} />
-                    <input type="hidden" name="__action" value="invite_student" />
-                    <button type="submit" class="btn btn-sm btn-outline-primary">
-                      <i class="fas fa-user-plus me-1" aria-hidden="true" /> Re-invite student
-                    </button>
-                  </form>
-                )}
-              </div>
+    <TimezoneContext.Provider value={course_instance.display_timezone}>
```

Comment:
Clever. Too clever? I suppose I lean towards explicitly passing things down a lever 🤷 seems like it's way too easy to end up silently formatting things in UTC this way.

Reply:
This still blows up if no context is found and there is no value passed.

Reply:
(So you can't silently format in UTC)

Reply:
Nice, makes sense.